spring-batch-integration/src/main/java/org/springframework/batch/integration/partition/RemotePartitioningWorkerStepBuilder.java

```diff
@@ -0,0 +1,249 @@
+	 */
+	public RemotePartitioningWorkerStepBuilder beanFactory(BeanFactory beanFactory) {
+		Assert.notNull(beanFactory, "beanFactory must not be null");
+		this.beanFactory = beanFactory;
+		return this;
+	}
+
+	@Override
+	public RemotePartitioningWorkerStepBuilder repository(JobRepository jobRepository) {
+		super.repository(jobRepository);
+		return this;
+	}
+
+	@Override
+	public RemotePartitioningWorkerStepBuilder transactionManager(PlatformTransactionManager transactionManager) {
+		super.transactionManager(transactionManager);
+		return this;
+	}
+
+	@Override
+	public RemotePartitioningWorkerStepBuilder startLimit(int startLimit) {
+		super.startLimit(startLimit);
+		return this;
+	}
+
+	@Override
+	public RemotePartitioningWorkerStepBuilder listener(Object listener) {
+		super.listener(listener);
+		return this;
+	}
+
+	@Override
+	public RemotePartitioningWorkerStepBuilder listener(StepExecutionListener listener) {
+		super.listener(listener);
+		return this;
+	}
+
+	@Override
+	public RemotePartitioningWorkerStepBuilder allowStartIfComplete(boolean allowStartIfComplete) {
+		super.allowStartIfComplete(allowStartIfComplete);
+		return this;
+	}
+
+	@Override
+	public TaskletStepBuilder tasklet(Tasklet tasklet) {
+		configureWorkerIntegrationFlow();
+		return super.tasklet(tasklet);
+	}
+
+	@Override
+	public <I, O> SimpleStepBuilder<I, O> chunk(int chunkSize) {
+		configureWorkerIntegrationFlow();
+		return super.chunk(chunkSize);
+	}
+
+	@Override
+	public <I, O> SimpleStepBuilder<I, O> chunk(CompletionPolicy completionPolicy) {
+		configureWorkerIntegrationFlow();
+		return super.chunk(completionPolicy);
+	}
+
+	@Override
+	public PartitionStepBuilder partitioner(String stepName, Partitioner partitioner) {
+		configureWorkerIntegrationFlow();
+		return super.partitioner(stepName, partitioner);
+	}
+
+	@Override
+	public PartitionStepBuilder partitioner(Step step) {
+		configureWorkerIntegrationFlow();
+		return super.partitioner(step);
+	}
+
+	@Override
+	public JobStepBuilder job(Job job) {
+		configureWorkerIntegrationFlow();
+		return super.job(job);
+	}
+
+	@Override
+	public FlowStepBuilder flow(Flow flow) {
+		configureWorkerIntegrationFlow();
+		return super.flow(flow);
+	}
+
+	/**
+	 * Create an {@link IntegrationFlow} with a {@link StepExecutionRequestHandler}
+	 * configured as a service activator listening to the input channel and replying
+	 * on the output channel.
+	 */
+	private void configureWorkerIntegrationFlow() {
+		Assert.notNull(this.inputChannel, "An InputChannel must be provided");
+		Assert.notNull(this.jobExplorer, "A JobExplorer must be provided");
+
+		if (this.stepLocator == null) {
+			BeanFactoryStepLocator beanFactoryStepLocator = new BeanFactoryStepLocator();
+			beanFactoryStepLocator.setBeanFactory(this.beanFactory);
+			this.stepLocator = beanFactoryStepLocator;
+		}
+		if (this.outputChannel == null) {
+			this.outputChannel = new NullChannel();
```

Comment:
A debug log message here indicating that the null channel has been configured and the master must poll for results may be a good idea.

Reply:
Good idea. Added a debug message.